core/src/main/java/org/elasticsearch/index/query/MatchQueryParser.java

```diff
@@ -94,7 +94,8 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
                         }
                     } else if ("analyzer".equals(currentFieldName)) {
                         String analyzer = parser.text();
-                        if (parseContext.analysisService().analyzer(analyzer) == null) {
+                        //norelease we have to delay this check to the shard when parsing on the coordinating node
```

Comment:
yes I think we can remove all of these comments for queries that are not refactored yet. And we shoud address now similar comments for queries that are already refactored.